### ISSUE TITLE
[BUG]: Error after resetting a card 

### ISSUE THREAD

**Issue by Kukla11** [ankidroid/Anki-Android#16734]
### Checked for duplicates?

- [X] This issue is not a duplicate

### Does it also happen in the desktop version?

- [ ] This bug does not occur in the latest version of Anki

### What are the steps to reproduce this bug?

A card automatically marked as leech and I'm given the next one, I click Undo, then Reschedule > Reset progress > Restore original position when possible, now I see the leach card again. At this point, no matter what I choose (Again, Hard, Good, Easy) the error pops up (attached). 
![Screenshot_20240714_090947_AnkiDroid](https://github.com/user-attachments/assets/407a40ca-aa3b-45e4-b160-44fbade70131)

### Expected behaviour

The original position should have been restored and there must be a chance to choose any option (Again, Hard, Good, Easy) without running into the error. 

### Actual behaviour

The error appears. 

### Debug info

```text
AnkiDroid Version = 2.18.4 (f867da96b8304edaebd0361326316bc7514f8cde)

Backend Version = 0.1.38-anki24.04.1 (24.04.1 ccd9ca1a8309b80bcb50ddc5d99c7ce63440bce9)

Android Version = 14 (SDK 34)

ProductFlavor = play

Manufacturer = samsung

Model = SM-A528B

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 14; SM-A528B Build/UP1A.231005.007; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/126.0.6478.134 Mobile Safari/537.36

ACRA UUID = 81e61918-8bae-4d03-9899-af53090d381b

FSRS Enabled = false

Crash Reports Enabled = true
```

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by MinusMallard** [comment]
@Kukla11 hey can you share a video , I'm not able to regenerate this bug

**Comment by Kukla11** [comment]
> @Kukla11 hey can you share a video , I'm not able to regenerate this bug

Sure! https://vimeo.com/985633165/9ca2df1543

**Comment by AffanShaikhsurab** [comment]
@david-allison i would like to work on it , can you assign me the issue

**Comment by jainv4156** [comment]
Hi @david-allison , I successfully reproduced the bug in version 2.18.4. However, the bug does not appear in the 2.19.1-beta4-debug build. This suggests the issue might already be resolved in the latest code, but I think it’s still worth investigating to confirm.

**Comment by david-allison** [comment]
@jainv4156 what does further investigation entail?

**Comment by Meetdobariya6868** [comment]
i want to work in this, make sure i able to implement this. give me assign work and where to push this work ?

**Comment by david-allison** [comment]
@Meetdobariya6868 Can you reproduce this issue?

**Comment by AffanShaikhsurab** [comment]
you can follow the above steps and reproduce the error : 

![Image](https://github.com/user-attachments/assets/f98d0806-36c6-4c05-b8df-e5dd73c0d12d)

**Comment by AffanShaikhsurab** [comment]
yes the error still occurs 

![Image](https://github.com/user-attachments/assets/9cae129b-ea08-482f-824a-cb4c95053cf0)

**Comment by mikehardy** [comment]
@AffanShaikhsurab how did you get a card to automatically be marked as leech? Did you set up your deck options specifically to reproduce this with a low leech setting or similar? The exact steps to reproduce from a fresh app install with no cards would be helpful I think as we appear to have reproduction difficulties here but you appear to have nailed it

**Comment by ShaanNarendran** [comment]
Hi, I wanted to tackle the issue but on trying to reproduce it for the last 2 hours, I've not been able to. I've made around 25 attempts doing different things like using different leech thresholds. Since the card has to be in review, I tried setting the due date using the card browser to 0 so that I can get to the leech threshold and I also tried to change system time to do the same. In all of the attempts, I haven't been able to reproduce the bug and it works as expected. I'm still new to the process so if anyone has any suggestions on what I can try, I'm willing to look at it. I haven't tried using the same version the initial user used since I felt that if there's no issue in the newest release I could just report that first. Sorry for the message block, open to suggestions.

**Comment by david-allison** [comment]
@ShaanNarendran could you zip and upload a sample collection where the next failure would cause a leech? 

Using this export, if you use 2.18.4, can you reproduce the issue?

 https://github.com/ankidroid/Anki-Android/releases/v2.18.4

**Comment by ShaanNarendran** [comment]
> [@ShaanNarendran](https://github.com/ShaanNarendran) could you zip and upload a sample collection where the next failure would cause a leech?
> 
> Using this export, if you use 2.18.4, can you reproduce the issue?
> 
> https://github.com/ankidroid/Anki-Android/releases/v2.18.4

Sure, I haven't exported on ankidroid before but I attached what I assume is the collection zip. The settings I used are leech threshold at 2 and 2 cards in the deck, card A will be leeched on the next review. Let me try it on 2.18.4 to see if I can reproduce and update back.

[collection-20251206220622.zip](https://github.com/user-attachments/files/23990377/collection-20251206220622.zip)

**Comment by ShaanNarendran** [comment]
@david-allison Yep, so as soon as I tried it on the old version, the bug was reproduced. It's very simple to do so, with the collection I uploaded just go to card browser then select both cards, set due date to 0, once you go back to review you can click hard for A which will leech it and then follow the steps mentioned in the main issue post and you'll get the error. This definitely does not happen in the newest version however, I did alot of testing and the error never popped up.

![Image](https://github.com/user-attachments/assets/5a9ca6ae-dc69-48ef-9ee7-91f9b8b04ccb)

**Comment by david-allison** [comment]
Fantastic, one bug resolved. Thank you so much!!

**Comment by david-allison** [comment]
Fantastic, one bug resolved. Thank you so much!!

**Comment by david-allison** [comment]
Fantastic, one bug resolved. Thank you so much!!

**Comment by david-allison** [comment]
Fantastic, one bug resolved. Thank you so much!!

**Comment by ShaanNarendran** [comment]
> Fantastic, one bug resolved. Thank you so much!!

Glad to help out, thank you as well for the suggestion!
